### PR TITLE
[Fix] Fixed device_id in tools/test.py for the CPU.

### PR DESCRIPTION
**Pull request by SemyonBevzuk**
## Motivation
It was found that `tools/test.py` with argument `--device cpu` crashes with the following error:
```bash
2022-01-11 01:45:25,318 - mmdeploy - INFO - Successfully loaded onnxruntime custom ops from             /home/sbevzyuk/mmdeployment_dir/MMDeploy/build/lib/libmmdeploy_onnxruntime_ops.so
Traceback (most recent call last):
  File "/home/sbevzyuk/mmdeployment_dir/MMDeploy/tools/test.py", line 141, in <module>
    main()
  File "/home/sbevzyuk/mmdeployment_dir/MMDeploy/tools/test.py", line 112, in main
    model = MMDataParallel(model, device_ids=[device_id])
  File "/home/sbevzyuk/anaconda3/envs/mmdeployment/lib/python3.7/site-packages/mmcv/parallel/data_parallel.py", line 35, in __init__
    super(MMDataParallel, self).__init__(*args, dim=dim, **kwargs)
  File "/home/sbevzyuk/anaconda3/envs/mmdeployment/lib/python3.7/site-packages/torch/nn/parallel/data_parallel.py", line 140, in __init__
    self.src_device_obj = torch.device(device_type, self.device_ids[0])
RuntimeError: Device index must not be negative`
```
This PR proposes a fix for this problem. 

## Modification
- `mmdeploy/backend/openvino/wrapper.py`: Fixed work with the dictionary for Python 3.8.
- `mmdeploy/utils/device.py`: The documentation of the `parse_device_id` function was updated.
- `tests/test_utils/test_util.py`: Added tests for function `parse_device_id`.
- `tools/test.py`: Added fix for `cpu`.

### CODE REVIEW

File: tools/test.py [hunk 1]
```diff
@@ -104,7 +104,10 @@ def main():
     # load the model of the backend
     model = task_processor.init_backend_model(args.model)
 
-    device_id = parse_device_id(args.device)
+    if args.device == 'cpu':
+        device_id = 0
+    else:
+        device_id = parse_device_id(args.device)
```
Comment (AllentDan) [inline tools/test.py:110]: Seems `device_id` get set to 0 for both `cpu` and `cuda:0` device. Maybe we have pass None to device_ids in MMDataParallel like described [here](https://github.com/open-mmlab/mmcv/blob/b57825f3f1f38cf6033e0d6d36af644cc57465fa/mmcv/parallel/data_parallel.py#L28). Besides, `with_sync` also need to be taken into account.

File: mmdeploy/utils/device.py [hunk 2]
```diff
@@ -11,6 +11,7 @@ def parse_device_id(device: str) -> int:
 
     Returns:
         int: The parsed device id, defaults to `0`.
+            If device is 'cpu' it will return `-1`.
     """
     if device == 'cpu':
```
Comment (AllentDan) [inline mmdeploy/utils/device.py:16]: The docstring may get updated. `parse_device_id` parse both cpu and gpu devices

File: tools/test.py (hunk 1 above)
Comment (SemyonBevzuk) [reply tools/test.py]: Thanks, fixed.

File: mmdeploy/utils/device.py (hunk 2 above)
Comment (SemyonBevzuk) [reply mmdeploy/utils/device.py:20]: Updated the documentation and implementation a bit.

**Review by AllentDan — APPROVED** [review]
LGTM

**Review by grimoire — APPROVED** [review]
LGTM

**Review by RunningLeon — APPROVED** [review]
LGTM

**Pull request merged**